core/domain/param_domain.py

```diff
@@ -173,12 +203,19 @@ def from_dict(cls, param_change_dict):
             param_change_dict['customization_args']
         )
 
-    def get_value(self, context_params):
+    # The method `generate_value()` accepts first argument of type Any and it
+    # also returns values of type Any. So that's why argument context_params
+    # and the return type of `get_value()` is annotated as Any.
+    def get_value(self, context_params: Any) -> Any:
         """Generates a single value for a parameter change."""
+        # The second argument of method `generate_value()` can only accept
+        # values of type Dict[str, Any], but here we are providing
+        # self.customization_args which is of type CustomizationArgsDict. Thus
+        # to avoid MyPy error, we added an ignore here.
```

Comment:
I think `cast` is not applicable here because:
1. If we do `cast(Dict[str, Any], self.customization_args)` then it only changes the type of `self.customization_args` but below we are doing `**self.customization_args` which is of type `**CustomizationArgsDict`.  Then in that case it contradicts the type again. 
2. If we do `cast(Dict[str, Any], **self.customization_args)` the MYPY throws error that too many arguments are provided which makes sense.

Reply:
1. So `self.customization_args = cast(Dict[str, Any], self.customization_args)`, does not work?

Reply:
Thanks, Done!. 
But instead of `self.customization_args = cast(Dict[str, Any], self.customization_args)` i used `custom_args = cast(Dict[str, Any], self.customization_args)`. because the former one throwing 2 MyPy errors ( self.customization_args is read_only property,  assigning Dict[str, Any] to CustomizationArgsDict type variable).